tools/network/cloudflare/cloudflare.go

```diff
@@ -295,6 +301,37 @@ func (c *Cred) GetZones(ctx context.Context) (zones []Zone, err error) {
 	return zones, err
 }
 
+// GetZoneID returns a zoneID that matches the requested zoneDomainName.
+func (c *Cred) GetZoneID(ctx context.Context, zoneDomainName string) (zoneID string, err error) {
+	zoneID = ""
+	zones, err := c.GetZones(ctx)
+	if err != nil {
+		return
+	}
+	if len(zones) == 0 {
+		err = ErrUserNotPermitted
+		return
+	}
+	zoneDomainName = strings.ToLower(zoneDomainName)
+	var matchingZone *Zone
+	for _, zone := range zones {
+		if zoneDomainName == strings.ToLower(zone.DomainName) {
+			// found a match.
+			if matchingZone != nil {
+				// we already had a previous match ?!
+				err = ErrInvalidDomainName
+				return
+			}
+			matchingZone = &zone
+		}
+	}
+	if matchingZone == nil {
+		err = ErrUserNotPermitted
```

Comment:
This seems like a misleading error message.  Something more like 'no zones matching %s for specified credentials'?

Reply:
np